octavia_f5/controller/worker/controller_worker.py

```diff
@@ -89,21 +89,74 @@ def __init__(self):
     def pending_sync(self):
         """
         Reconciliation loop that
-        - schedules unscheduled load balancers to this worker
+        - synchronizes load balancers that are in a PENDING state
         - deletes load balancers that are PENDING_DELETE
+        - executes a full sync on F5 devices that were offline but are now back online
         """
 
-        # schedule unscheduled load balancers to this worker
+        # synchronize load balancers that are in a PENDING state
         self.sync_loadbalancers()
 
         # delete load balancers that are PENDING_DELETE
+        session = db_apis.get_session()
         lbs_to_delete = self._loadbalancer_repo.get_all_from_host(
-            db_apis.get_session(),
-            provisioning_status=lib_consts.PENDING_DELETE)
+            session, provisioning_status=lib_consts.PENDING_DELETE)
         for lb in lbs_to_delete:
             LOG.info("Found pending deletion of lb %s", lb.id)
             self.delete_load_balancer(lb.id)
 
+        # Execute a full sync on F5 devices that were offline but are now back online.
+        self.full_sync_reappearing_devices(session)
+
+    @lockutils.synchronized('tenant_refresh')
+    def full_sync_reappearing_devices(self, session):
+        """Execute a full sync on F5 devices that were offline but are now back online. These devices have
+        status='BOOTING' (set by status manager) in their corresponding amphora table entry. """
+        for device in self.bigip.bigips:
+            device_name = device.hostname
+            device_entry = self._amphora_repo.get(session,
+                                         status=constants.AMPHORA_BOOTING,
+                                         compute_flavor=CONF.host,
+                                         load_balancer_id=None,
+                                         cached_zone=device_name)
+            if not device_entry:
+                continue
+            LOG.info("Device reappeared: %s. Doing a full sync.", device_name)
+
+            # get all load balancers (of this host)
+            lbs = self._loadbalancer_repo.get_all_from_host(session)
+
+            # Change the BigIP temporarily for refreshing load balancers
+            originally_active_bigip = self.bigip.active_bigip
+            self.bigip.active_bigip = device
```

Comment:
This is too error prone, I would rather like to see two instances of the as3restclient.

Reply:
I was counting on 'tenant_refresh' sufficing. However, forgetting to acquire that lock when writing/changing functions in the future could lead to hardly debuggable behavior, sooooo..... yea, let's just reinstanciate `as3restclient` :)